test/yargs-parser.js

```diff
@@ -762,6 +762,29 @@ describe('yargs-parser', function () {
       argv.should.have.property('n', 123)
     })
 
+    it('should set n to the numeric value 123, with n at the end of a group', function () {
+      var argv = parser([ '-ab5n123' ])
```

Comment:
This is an odd edge case. I would expect something like `-x1` to be parsed as `x: 1`, which I think is common among most POSIX bins, but I don't know what the typical interpretation of `-x1y5` is.

In this test case, I would expect the result to be one of the following:

1. `a: true`, `b: 5`, `n: 123`
2. `a: true`, `b: true`, `5: true`, `n: 123`